main.go

```diff
@@ -139,6 +143,14 @@ func main() {
 		os.Exit(1)
 	}
 
+	if enableprofile {
+		mgr.AddMetricsExtraHandler("/debug/pprof/", http.HandlerFunc(pprof.Index))
```

Comment:
We need to check the returned error here and for the other mgr.AddMetricsExtraHandler cases.
```
err = mgr.AddMetricsExtraHandler("/debug/pprof/", http.HandlerFunc(pprof.Index))
if err != nil {
	setupLog.Error(err, "unable to attach pprof to webserver")
	os.Exit(1)
}
```